include/common/core/freertos_task.hpp

```diff
@@ -73,4 +74,28 @@ class FreeRTOSTask {
     std::function<void()> starter{};
 };
 
+template <uint32_t StackDepth,
+          template <template <typename> typename> typename TaskObj>
```

Comment:
one liner comment here to explain that `template <template <typename> typename> typename TaskObj` part of the template parameter list takes care of `template <template <class> class QueueImpl>` (or whatever it actually maps to) sounds helpful to me.

Reply:
Yeah that's a good idea, will do